packages/browser-agent-core/common/wrap/wrap-events.js

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {ee as baseEE} from '../event-emitter/contextual-ee'
+import { createWrapperWithEmitter as wfn } from './wrap-function'
+import { getOrSet } from '../util/get-or-set'
+
+const wrapped = {}
+
+export function wrapEvents(sharedEE) {
+  var ee = scopedEE(sharedEE)
+  if (wrapped[ee.debugId]) return ee
+  wrapped[ee.debugId] = true
+  var wrapFn = wfn(ee, true)
+
+  var XHR = XMLHttpRequest
+  var ADD_EVENT_LISTENER = 'addEventListener'
+  var REMOVE_EVENT_LISTENER = 'removeEventListener'
+
+  // Guard against instrumenting environments w/o necessary features
+  if ('getPrototypeOf' in Object) {
+    findAndWrapNode(document)
+    findAndWrapNode(window)
+    findAndWrapNode(XHR.prototype)
+    // eslint-disable-next-line
+  } else if (XHR.prototype.hasOwnProperty(ADD_EVENT_LISTENER)) {
+    wrapNode(window)
+    wrapNode(XHR.prototype)
+  }
+
+  ee.on(ADD_EVENT_LISTENER + '-start', function (args, target) {
+    var originalListener = args[1]
+    if (originalListener === null ||
+      (typeof originalListener !== 'function' && typeof originalListener !== 'object')
+    ) {
+      return
+    }
+
+    var wrapped = getOrSet(originalListener, 'nr@wrapped', function () {
+      var listener = {
+        object: wrapHandleEvent,
+        'function': originalListener
+      }[typeof originalListener]
+
+      return listener ? wrapFn(listener, 'fn-', null, (listener.name || 'anonymous')) : originalListener
+
+      function wrapHandleEvent() {
+        if (typeof originalListener.handleEvent !== 'function') return
+        return originalListener.handleEvent.apply(originalListener, arguments)
+      }
+    })
+
+    this.wrapped = args[1] = wrapped
+  })
+
+  ee.on(REMOVE_EVENT_LISTENER + '-start', function (args) {
+    args[1] = this.wrapped || args[1]
+  })
+
+  function findAndWrapNode(object) {
+    var step = object
+    // eslint-disable-next-line
+    while (step && !step.hasOwnProperty(ADD_EVENT_LISTENER)) { step = Object.getPrototypeOf(step) }
+    if (step) { wrapNode(step) }
+  }
+
+  function wrapNode(node) {
+    wrapFn.inPlace(node, [ADD_EVENT_LISTENER, REMOVE_EVENT_LISTENER], '-', uniqueListener)
+  }
+
+  function uniqueListener(args, obj) {
+    // Context for the listener is stored on itself.
+    return args[1]
+  }
+
+  return ee
+}
+
+export function scopedEE(sharedEE){
```

Comment:
I only see this `export function scopedEE` pattern used as an import for one test, and I'm not sure what it's doing.

I see scopedEE imported in this test, but I think the test needs to be changed to import and call wrapEvents

https://github.com/newrelic/newrelic-browser-agent/blob/7e13aea02cc7ac928592bfaf71e5a6f0d18bac23/tests/browser/add-event-listener-options.browser.js#L8-L10

Is there any other use for this extra export on all the wrap-* files?

Reply:
Yeah you're right; good spotting. Same as the add-event-listener.

Reply:
only thing it does is shortcut accessing the namespaced ee without actually wrapping anything (and without needing to know which namespace belongs to which wrap module), (ie. ee.get('events'), ee.get('timings), etc).  Definitely not needed.